packages/assets-controllers/src/TokenBalancesController.test.ts

```diff
@@ -1,191 +1,250 @@
   });
 
   it('should poll and update balances in the right interval', async () => {
-    const mock = sinon.stub(
+    const updateBalancesSpy = jest.spyOn(
       TokenBalancesController.prototype,
       'updateBalances',
     );
-    new TokenBalancesController(
-      {
-        onTokensStateChange: sinon.stub(),
-        getSelectedAddress: () => '0x1234',
-        getERC20BalanceOf: sinon.stub(),
-      },
-      { interval: 10 },
-    );
-    expect(mock.called).toBe(true);
-    expect(mock.calledTwice).toBe(false);
 
-    await advanceTime({ clock, duration: 15 });
-    expect(mock.calledTwice).toBe(true);
+    new TokenBalancesController({
+      interval: 10,
+      onTokensStateChange: jest.fn(),
+      getSelectedAddress: () => '0x1234',
+      getERC20BalanceOf: jest.fn(),
+      messenger: getMessenger(),
+    });
+    await flushPromises();
+
+    expect(updateBalancesSpy).toHaveBeenCalled();
+    expect(updateBalancesSpy).not.toHaveBeenCalledTimes(2);
+
+    jest.advanceTimersByTime(15);
+
+    expect(updateBalancesSpy).toHaveBeenCalledTimes(2);
   });
 
-  it('should not update rates if disabled', async () => {
-    const tokenBalances = new TokenBalancesController(
-      {
-        onTokensStateChange: sinon.stub(),
-        getSelectedAddress: () => '0x1234',
-        getERC20BalanceOf: sinon.stub(),
-      },
-      {
-        disabled: true,
-        interval: 10,
+  it('should not update banlances if disabled', async () => {
+    const address = '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0';
+    const controller = new TokenBalancesController({
+      disabled: true,
+      tokens: [{ address, decimals: 18, symbol: 'EOS', aggregators: [] }],
+      interval: 10,
+      onTokensStateChange: jest.fn(),
+      getSelectedAddress: () => '0x1234',
+      getERC20BalanceOf: jest.fn().mockReturnValue(new BN(1)),
+      messenger: getMessenger(),
+    });
+
+    await controller.updateBalances();
+
+    expect(controller.state.contractBalances).toStrictEqual({});
+  });
+
+  it('should update banlances if controller is manually enabled', async () => {
+    const address = '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0';
+    const tokenBalances = new TokenBalancesController({
+      disabled: true,
+      tokens: [{ address, decimals: 18, symbol: 'EOS', aggregators: [] }],
+      interval: 10,
+      onTokensStateChange: jest.fn(),
+      getSelectedAddress: () => '0x1234',
+      getERC20BalanceOf: jest.fn().mockReturnValue(new BN(1)),
+      messenger: getMessenger(),
+    });
+
+    await tokenBalances.updateBalances();
+
+    expect(tokenBalances.state.contractBalances).toStrictEqual({});
+
+    tokenBalances.enable();
+    await tokenBalances.updateBalances();
+
+    expect(tokenBalances.state.contractBalances).toStrictEqual({
+      '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0': '1',
+    });
+  });
+
+  it('should not update banlances if controller is manually disabled', async () => {
```

Comment:
if we add `"should not update balances if controller is manually disabled"`, it would be:

```
it('should not update balances if controller is manually disabled', async () => {
    const address = '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0';
    const controller = new TokenBalancesController({
      disabled: false,
      tokens: [{ address, decimals: 18, symbol: 'EOS', aggregators: [] }],
      interval: 10,
      onTokensStateChange: jest.fn(),
      getSelectedAddress: () => '0x1234',
      getERC20BalanceOf: jest.fn().mockReturnValue(new BN(1)),
      messenger: getMessenger(),
    });
    
    await controller.updateBalances();

    expect(controller.state.contractBalances).toStrictEqual({
      '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0': '1',
    });

    controller.disable();
    await controller.updateBalances();

    expect(controller.state.contractBalances).toStrictEqual({
      '0x86fa049857e0209aa7d9e616f7eb3b3b78ecfdb0': '1',
    });
});
```

  However as the constructor calls updateBalances -> this means contractBalances well have the passed value from the constructor then calling updateBalances after disabling controller and not changes in tokens would make no change compared to keep in it enabled.